### PR TITLE
Add episode on releases and tags

### LINKED ISSUE #7: Write about using releases and versions

### PR DESCRIPTION
**Pull request by alexdewar**
Hi all,

Here's the first draft of my releases/tags section. Let me know if you have any suggestions!

Closes #7.

### CODE REVIEW

File: _episodes/l1-06-releases_tags.md [hunk 1]
```diff
@@ -3,19 +3,355 @@ title: "Code versions, releases and tags"
 teaching: 0 # TBD
 exercises: 0 # TBD
 questions:
+- "What is a Git tag and how does it differ from a branch?"
 - "How and when should I release a new version of my code?"
 - "What is the difference between major and minor version changes?"
-- "How can I effectively communicate what has changed between code version?"
-
+- "How can I effectively communicate what has changed between versions?"
+- "How can I tag commits?"
```
Comment (dalonsoa) [inline _episodes/l1-06-releases_tags.md:10]: Maybe move this up, to keep a more logical order?

File: _episodes/l1-06-releases_tags.md [hunk 2]
```diff
@@ -3,19 +3,355 @@ title: "Code versions, releases and tags"
...
+for a future date by giving it the tag "tasty".  You can do this like so:
+
+~~~
+$ git tag tasty [commit hash]
+~~~
+
+In my case, I ran:
+
+~~~
+$ git tag tasty 366f4b51e661fd96a9bde001138886a4216b7da1
```
Comment (dalonsoa) [inline _episodes/l1-06-releases_tags.md:150]: Generally, we use the short versions of the tags. I assume that works also here, right?

**Review by dalonsoa — CHANGES_REQUESTED** [review]
I looks really nice and it is easy to follow and understand. My only comment is that there are no proper exercises. You do encourage to code along and follow the commands you are running, but you don't stop, give them time to do their stuff and then check the results. Even though most tasks are simple, it will be worth to have them arranged as exercises.

File: _episodes/l1-06-releases_tags.md [hunk 3]
```diff
@@ -3,19 +3,355 @@ title: "Code versions, releases and tags"
...
+colleague), remember that you can always obtain the current commit hash like so:
+
+~~~sh
+$ git rev-parse --short HEAD
+~~~
+{: .commands}
+~~~
+047e4fe
+~~~
+{: .output}
```
Comment (TinyMarsh) [inline _episodes/l1-06-releases_tags.md:73]: I would consider moving the "When *don't* you need..." section below the "When should I consider..." section. It's subjective, but for me it reads better.

**Review by TinyMarsh — CHANGES_REQUESTED** [review]
I really like this. The content is explained nicely and I like the injection of humour :) I agree with Diego; some exercises wouldn't go amiss. Admittedly, I'm mainly thinking about that in terms of how it would feel to actually deliver the course. It's nice to break up periods of talking with a "now you have a go".

File: _episodes/l1-06-releases_tags.md (hunk 2 above)
Comment (alexdewar) [reply _episodes/l1-06-releases_tags.md:150]: Yup! Done

File: _episodes/l1-06-releases_tags.md (hunk 1 above)
Comment (alexdewar) [reply _episodes/l1-06-releases_tags.md:10]: Good point.

File: _episodes/l1-06-releases_tags.md (hunk 3 above)
Comment (alexdewar) [reply _episodes/l1-06-releases_tags.md:73]: Good idea

**Comment by alexdewar** [comment]
> I looks really nice and it is easy to follow and understand. My only comment is that there are no proper exercises. You do encourage to code along and follow the commands you are running, but you don't stop, give them time to do their stuff and then check the results. Even though most tasks are simple, it will be worth to have them arranged as exercises.

Good suggestion. I've added two exercises, but nothing obvious presented itself to me for the other sections. Any ideas? @TinyMarsh?

**Review by dalonsoa — APPROVED** [review]
All good! Merging!

**Comment by dalonsoa** [comment]
@dandavies99 , could you have a quick look at this lesson, and then we move on?

**Comment by dalonsoa** [comment]
And @TinyMarsh

**Review by TinyMarsh — APPROVED** [review]
This looks good to me.

**Pull request merged**